galley/pkg/runtime/state.go

```diff
@@ -0,0 +1,154 @@
+//  Copyright 2018 Istio Authors
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+package runtime
+
+import (
+	"fmt"
+	"sync"
+
+	"github.com/gogo/protobuf/proto"
+	"github.com/gogo/protobuf/types"
+
+	"istio.io/api/config/mcp/v1alpha1"
+	"istio.io/istio/galley/pkg/mcp/snapshot"
+	"istio.io/istio/galley/pkg/runtime/resource"
+)
+
+// State is the in-memory state of Galley.
+type State struct {
+	schema *resource.Schema
+
+	// version counter is a nonce that generates unique ids for each updated view of State.
+	versionCounter int64
+
+	// entries for per-kind State.
+	entries     map[resource.Kind]*kindState
+	entriesLock sync.Mutex
```

Comment:
nit: I believe the idiom is to declare the mutex before the fields being protected by the mutex.